app/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDao.java

```diff
@@ -93,19 +119,63 @@ public long getNumberOfParticipants(StudyIdentifier studyIdentifier) {
 
-        // We need the count of unique study participants, users may end up signing 
+        // We need the count of unique study participants, users may end up signing
         // more than one version of the consent.
         Set<String> healthCodes = Sets.newHashSet();
-        List<DynamoUserConsent2> mappings = mapper.scan(DynamoUserConsent2.class, scan);
-        for (DynamoUserConsent2 consent : mappings) {
-            healthCodes.add(consent.getHealthCode());
-        }
+        List<DynamoUserConsent2> mappings = mapperV2.scan(DynamoUserConsent2.class, scan);
+        mappings.stream().forEach(consent -> healthCodes.add(consent.getHealthCode()));
         return healthCodes.size();
     }
 
-    private DynamoUserConsent2 getUserConsent(String healthCode, StudyConsent studyConsent) {
-        DynamoUserConsent2 consent = new DynamoUserConsent2(healthCode, studyConsent);
-        consent = mapper.load(consent);
-        return consent;
+    @Override
+    public void deleteConsentRecords(String healthCode, StudyIdentifier studyIdentifier) {
+        List<DynamoUserConsent3> consentsV3 = getAllUserConsentRecords3(healthCode, studyIdentifier.getIdentifier());
+        if (!consentsV3.isEmpty()) {
+            List<FailedBatch> failures = mapperV3.batchDelete(consentsV3);
+            BridgeUtils.ifFailuresThrowException(failures);
+        }
+        DynamoUserConsent2 consentV2 = getUserConsent2(healthCode, studyIdentifier.getIdentifier());
+        if (consentV2 != null) {
+            mapperV2.delete(consentV2);    
+        }
+    }
+    
+    @Override
+    public boolean migrateConsent(String healthCode, StudyIdentifier studyIdentifier) {
+        DynamoUserConsent2 consent2 = getUserConsent2(healthCode, studyIdentifier.getIdentifier());
+        DynamoUserConsent3 consent3 = getUserConsent3(healthCode, studyIdentifier.getIdentifier());
+
+        if (consent2 != null && consent3 == null) {
+            DynamoUserConsent3 newConsent = new DynamoUserConsent3();
+            newConsent.setConsentCreatedOn(consent2.getConsentCreatedOn());
+            newConsent.setHealthCodeStudy(healthCode, studyIdentifier.getIdentifier());
+            newConsent.setSignedOn(consent2.getSignedOn());
+            mapperV3.save(newConsent);
+            return true;
+        }
+        return false;
+    }
+
+    private DynamoUserConsent2 getUserConsent2(String healthCode, String studyIdentifier) {
+        DynamoUserConsent2 hashKey = new DynamoUserConsent2(healthCode, studyIdentifier);
+
+        return mapperV2.load(hashKey);
+    }
+
+    private DynamoUserConsent3 getUserConsent3(String healthCode, String studyIdentifier) {
+        // This record has a range key so there are multiple consents. Get the first one, for now.
+        List<DynamoUserConsent3> consents = getAllUserConsentRecords3(healthCode, studyIdentifier);
+        
+        return (consents.isEmpty()) ? null : consents.get(0);
+    }
```

Comment:
Minor optimization: Query expressionwith .withScanIndexForward(false).withLimit(1). This will always get you the latest consent, and nothing more.